[16.0][IMP]helpdesk_mgmt: interface improvements

flotho opened the pull request:
Depends on : 

- [x]  https://github.com/OCA/helpdesk/pull/585
- [X]  https://github.com/OCA/helpdesk/pull/583
 
Backend Improvements : 

- [x] Add kanban state in multiple views
- [x] Reset kanban state when changing the stage 
- [x] Refactor the form view to be closer to task's one 
- [x] New ticket view with searchpanel and default filter

Portal improvements : 

- [x] Change submission button size in the home of the portal user 
- [x] Refactor the list view : 
- Default group by stage when reaching the list view
- Change columns to be more consistent with tasks
- Change the css on groups and lines to be more concise

flotho commented:
Purpose is to boost the user experience backend : 
![image](https://github.com/OCA/helpdesk/assets/1043935/a2f1e3f9-c352-4217-ad25-81dd8280112f)

![image](https://github.com/OCA/helpdesk/assets/1043935/64ee872d-c47a-45c7-b82b-62b91c4573a7)


some improvements inspired from https://github.com/OCA/helpdesk/pull/556 . @eLBati how do you want me to mention it ?

flotho commented:
Regarding the portal, multiple things could be imporved as proposed there : https://github.com/OCA/helpdesk/pull/577
It looks difficut to proper alter/inherit `portal_docs_entry` to add the button as proposed by @pedrobaeza 
So my proposal is to use css to grow on all the line : 
![image](https://github.com/OCA/helpdesk/assets/1043935/8f5a1559-2388-4627-903b-e1e415bea268)

flotho commented:
Ive also changed the view list : 
![image](https://github.com/OCA/helpdesk/assets/1043935/655e3b89-6099-4b36-83be-0418ef0f28c6)

DorianMAG reviewed (approved):
Test in runboat,
In Helpdesk app, click on button Ticket to see all our tickets with default filter.

Thx for this worl
LGTM

JulienMartinez reviewed (approved):
test in the runboat,
everything works for me, 

thanks for the work
LGTM

eLBati commented:
> how do you want me to mention it ?

Hi @flotho you could cherry-pick the commit, if applicable

flotho commented:
Hi @eLBati , thanks, I'll do another try since dependencies have nbeen merged

flotho commented:
> > how do you want me to mention it ?
> 
> Hi @flotho you could cherry-pick the commit, if applicable

cherry pick done

flotho commented:
Hi @eLBati ,

Your review will be appreciated

Regards

flotho commented:
Hi @eLBati @DorianMAG  @gaelTorrecillas , @JulienMartinez , any review here would be appreciated

DorianMAG reviewed (approved):
Test in runboat,
In Helpdesk app, click on button Ticket to see all our tickets with default filter.

Thx for this work
LGTM

flotho commented:
Hi @marcelsavegnago , your review would be appreciated here as it impacts the portal

DorianMAG reviewed (approved):
Test in runboat,
In Helpdesk app, click on button Ticket to see all our tickets with default filter.

Thx for this work
LGTM

flotho commented:
pop @JulienMartinez , your review is expected here

JulienMartinez reviewed (approved):
test in the runboat,
everything works for me,

thanks for the work
LGTM

flotho commented:
hum....
@pedrobaeza ould you help me regarding the codecov?
Looks like only readme is concerned.
Do I have to reset the readme file ?

Regards

flotho commented:
hum, the codecov failed on a the commit https://github.com/OCA/helpdesk/commit/bc8e7407a66c01b0adf4cd4ef5e869534dc092cd that is part of a post merge commit

flotho committed:
[IMP]helpdesk_mgmt: Add internal notes

flotho commented:
> hum, the codecov failed on a the commit [bc8e740](https://github.com/OCA/helpdesk/commit/bc8e7407a66c01b0adf4cd4ef5e869534dc092cd) that is part of a post merge commit

sorry to buzz @pedrobaeza , how can I get out from this situation, the codecov regression has been generated by a robot commit.
In this case, what is the procedure ?
Regards

pedrobaeza commented:
Don't get mad with that, as:

1. Codecov check is not mandatory. It's just a metric, but it doesn't block the merge.
2. In my experience, it's not working well, giving false reds.

pedrobaeza commented on helpdesk_mgmt/__manifest__.py:
```diff
@@ -14,6 +14,7 @@
     "Domatix, "
     "Factor Libre, "
     "SDi Soluciones, "
+    "Mind And Go, "
```
I'm afraid these improvements don't qualify for co-authorship, just for contributors.

flotho replied:
thanks @pedrobaeza 
Done !

pedrobaeza commented:
Please squash 1st, 2nd and 4th commit together, as they are not adding value as separated.

flotho commented:
> Please squash 1st, 2nd and 4th commit together, as they are not adding value as separated.

done! thanks @pedrobaeza

flotho committed:
[IMP]helpdesk_mgmt adding kanban_state to form, tree and search views

[IMP]helpdesk_mgmt: Add kanban state in multiple views

[REF]helpdesk_mgmt: use template folders

[IMP]helpdesk_mgmt: Backend UI improvements

[IMP]helpdesk_mgmt: Reset kanban state at state change

[IMP]helpdesk_mgmt: new ticket button

[IMP]helpdesk_mgmt: Portal display improvements

[IMP]helpdesk_mgmt: ACL on tickt list

[LINT]helpdesk_mgmt

[LINT]helpdesk_mgmt

[LINT]helpdesk_mgmt

[IMP]helpdesk_mgmt: authoring

[LINT]helpdesk_mgmt

flotho commented:
no more conflicts !

pedrobaeza commented on helpdesk_mgmt/templates/helpdesk_ticket_templates.xml:
```diff
@@ -56,71 +50,86 @@
                     <p>No tickets found.</p>
                 </div>
             </t>
+            <div class="d-flex pt-1">
```
This change should go and debated in a separate commit.

flotho replied:
Thanks for the review  @pedrobaeza 
Considering your request and the fact that I squashed everything ;-) I think I will rewrite all this PR to fullfill your requirements .
Do you have additional advice you want me to isolate in a commit ?
Regards

pedrobaeza replied:
Yeah, when I said "squash the commit history a bit" was for avoiding noise, but grouping things in a logical way for being easily reviewed isolatedly. There's no absolute choice here.

pedrobaeza replied:
And if grouping together, the first line of the commit message should reflect everything the commit contains.
